BigImageViewer/src/main/java/com/github/piasy/biv/view/BigImageView.java

```diff
@@ -502,32 +525,81 @@ public void onAnimationRepeat(Animation animation) {
     }
 
     @UiThread
-    private void doShowImage(final int imageType, final File image) {
-        if (mMainView != null) {
-            removeView(mMainView);
-        }
+    private void doShowImage(final int imageType, final File image, final boolean useThumbnailView) {
 
-        mMainView = mViewFactory.createMainView(getContext(), imageType, image, mInitScaleType);
-        if (mMainView == null) {
-            onFail(new RuntimeException("Image type not supported: "
-                                        + ImageInfoExtractor.typeName(imageType)));
-            return;
-        }
+        if (useThumbnailView) {
+
+            if (mThumbnailView == null) {
+                mThumbnailView = mViewFactory.createThumbnailView(getContext(), mThumbnailScaleType);
+            }
+
+            if (mThumbnailView.getVisibility() != View.VISIBLE) {
+                mThumbnailView.setVisibility(View.VISIBLE);
+            }
+
+            if (mThumbnailView != null) {
 
-        addView(mMainView, ViewGroup.LayoutParams.MATCH_PARENT,
-                ViewGroup.LayoutParams.MATCH_PARENT);
-        mMainView.setOnClickListener(mOnClickListener);
-        mMainView.setOnLongClickListener(mOnLongClickListener);
+                if (!(mThumbnailView.getParent() == this)) {
+                    addView(mThumbnailView, ViewGroup.LayoutParams.MATCH_PARENT,
+                            ViewGroup.LayoutParams.MATCH_PARENT);
+                }
+
+                mThumbnailView.setOnClickListener(mOnClickListener);
+                mThumbnailView.setOnLongClickListener(mOnLongClickListener);
+
+                ((ImageView) mThumbnailView).setAdjustViewBounds(true);
```

Comment:
I think we'd better check if it's an ImageView here, because the `ImageViewFactory` may have some other implementation.

Reply:
done!